Errors when rescanning comics.

**Describe The Bug**
Whenever comics are rescanned, the backend throws an exception and the process fails.

**To Reproduce The Bug**
Steps to reproduce the behavior. Please be as complete as possible and describe the exact steps and values used to cause the error:
1. Select one or more comics.
1. Click on the **Rescan comics** button.
1. The rescan process eventually fails with an exception.

**Expected Behavior**
1. The comics should be rescanned without an exception.

**Additional Context**
```
org.springframework.dao.DataIntegrityViolationException: could not execute statement; SQL [n/a]; constraint ["PUBLIC.CONSTRAINT_A5_INDEX_1 ON PUBLIC.COMICFILEDETAILS(COMICBOOKID) VALUES 319"; SQL statement:
insert into ComicFileDetails (id, ComicBookId, FileHash) values (null, ?, ?) [23505-200]]; nested exception is org.hibernate.exception.ConstraintViolationException: could not execute statement
	at org.springframework.orm.jpa.vendor.HibernateJpaDialect.convertHibernateAccessException(HibernateJpaDialect.java:276) ~[spring-orm-5.3.15.jar!/:5.3.15]
	at org.springframework.orm.jpa.vendor.HibernateJpaDialect.translateExceptionIfPossible(HibernateJpaDialect.java:233) ~[spring-orm-5.3.15.jar!/:5.3.15]
	at org.springframework.orm.jpa.AbstractEntityManagerFactoryBean.translateExceptionIfPossible(AbstractEntityManagerFactoryBean.java:551) ~[spring-orm-5.3.15.jar!/:5.3.15]
	at org.springframework.dao.support.ChainedPersistenceExceptionTranslator.translateExceptionIfPossible(ChainedPersistenceExceptionTranslator.java:61) ~[spring-tx-5.3.15.jar!/:5.3.15]
	at org.springframework.dao.support.DataAccessUtils.translateIfNecessary(DataAccessUtils.java:242) ~[spring-tx-5.3.15.jar!/:5.3.15]
	at org.springframework.dao.support.PersistenceExceptionTranslationInterceptor.invoke(PersistenceExceptionTranslationInterceptor.java:152) ~[spring-tx-5.3.15.jar!/:5.3.15]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.3.15.jar!/:5.3.15]
	at org.springframework.data.jpa.repository.support.CrudMethodMetadataPostProcessor$CrudMethodMetadataPopulatingMethodInterceptor.invoke(CrudMethodMetadataPostProcessor.java:174) ~[spring-data-jpa-2.6.1.jar!/:2.6.1]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.3.15.jar!/:5.3.15]
	at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:97) ~[spring-aop-5.3.15.jar!/:5.3.15]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.3.15.jar!/:5.3.15]
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:215) ~[spring-aop-5.3.15.jar!/:5.3.15]
	at jdk.proxy2/jdk.proxy2.$Proxy179.save(Unknown Source) ~[na:na]
	at org.comixedproject.service.comicbooks.ComicBookService.onComicStateChange(ComicBookService.java:274) ~[comixed-services-1.1.0-1.jar!/:1.1.0-1]
	at org.comixedproject.state.comicbooks.ComicStateHandler$1.lambda$postStateChange$0(ComicStateHandler.java:73) ~[comixed-state-1.1.0-1.jar!/:1.1.0-1]
	at java.base/java.lang.Iterable.forEach(Iterable.java:75) ~[na:na]
	at org.comixedproject.state.comicbooks.ComicStateHandler$1.postStateChange(ComicStateHandler.java:73) ~[comixed-state-1.1.0-1.jar!/:1.1.0-1]
	at org.springframework.statemachine.support.StateMachineInterceptorList.postStateChange(StateMachineInterceptorList.java:118) ~[spring-statemachine-core-2.4.0.jar!/:2.4.0]
	at org.springframework.statemachine.support.AbstractStateMachine.callPostStateChangeInterceptors(AbstractStateMachine.java:890) ~[spring-statemachine-core-2.4.0.jar!/:2.4.0]
	at org.springframework.statemachine.support.AbstractStateMachine.entryToState(AbstractStateMachine.java:1273) ~[spring-statemachine-core-2.4.0.jar!/:2.4.0]
	at org.springframework.statemachine.support.AbstractStateMachine.entryToState(AbstractStateMachine.java:1263) ~[spring-statemachine-core-2.4.0.jar!/:2.4.0]
	at org.springframework.statemachine.support.AbstractStateMachine.setCurrentStateInternal(AbstractStateMachine.java:1059) ~[spring-statemachine-core-2.4.0.jar!/:2.4.0]
	at org.springframework.statemachine.support.AbstractStateMachine.setCurrentState(AbstractStateMachine.java:1033) ~[spring-statemachine-core-2.4.0.jar!/:2.4.0]
	at org.springframework.statemachine.support.AbstractStateMachine.switchToState(AbstractStateMachine.java:926) ~[spring-statemachine-core-2.4.0.jar!/:2.4.0]
	at org.springframework.statemachine.support.AbstractStateMachine.access$500(AbstractStateMachine.java:82) ~[spring-statemachine-core-2.4.0.jar!/:2.4.0]
	at org.springframework.statemachine.support.AbstractStateMachine$3.transit(AbstractStateMachine.java:337) ~[spring-statemachine-core-2.4.0.jar!/:2.4.0]
	at org.springframework.statemachine.support.DefaultStateMachineExecutor.handleTriggerTrans(DefaultStateMachineExecutor.java:287) ~[spring-statemachine-core-2.4.0.jar!/:2.4.0]
	at org.springframework.statemachine.support.DefaultStateMachineExecutor.handleTriggerTrans(DefaultStateMachineExecutor.java:210) ~[spring-statemachine-core-2.4.0.jar!/:2.4.0]
	at org.springframework.statemachine.support.DefaultStateMachineExecutor.processTriggerQueue(DefaultStateMachineExecutor.java:450) ~[spring-statemachine-core-2.4.0.jar!/:2.4.0]
	at org.springframework.statemachine.support.DefaultStateMachineExecutor.access$200(DefaultStateMachineExecutor.java:64) ~[spring-statemachine-core-2.4.0.jar!/:2.4.0]
	at org.springframework.statemachine.support.DefaultStateMachineExecutor$1.run(DefaultStateMachineExecutor.java:324) ~[spring-statemachine-core-2.4.0.jar!/:2.4.0]
	at org.springframework.core.task.SyncTaskExecutor.execute(SyncTaskExecutor.java:50) ~[spring-core-5.3.15.jar!/:5.3.15]
	at org.springframework.statemachine.support.DefaultStateMachineExecutor.scheduleEventQueueProcessing(DefaultStateMachineExecutor.java:353) ~[spring-statemachine-core-2.4.0.jar!/:2.4.0]
	at org.springframework.statemachine.support.DefaultStateMachineExecutor.execute(DefaultStateMachineExecutor.java:162) ~[spring-statemachine-core-2.4.0.jar!/:2.4.0]
	at org.springframework.statemachine.support.AbstractStateMachine.sendEventInternal(AbstractStateMachine.java:605) ~[spring-statemachine-core-2.4.0.jar!/:2.4.0]
	at org.springframework.statemachine.support.AbstractStateMachine.sendEvent(AbstractStateMachine.java:219) ~[spring-statemachine-core-2.4.0.jar!/:2.4.0]
	at org.comixedproject.state.comicbooks.ComicStateHandler.fireEvent(ComicStateHandler.java:121) ~[comixed-state-1.1.0-1.jar!/:1.1.0-1]
	at org.comixedproject.state.comicbooks.ComicStateHandler.fireEvent(ComicStateHandler.java:95) ~[comixed-state-1.1.0-1.jar!/:1.1.0-1]
	at org.comixedproject.batch.comicbooks.writers.AbstractComicWriter.lambda$write$0(AbstractComicWriter.java:51) ~[comixed-batch-1.1.0-1.jar!/:1.1.0-1]
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511) ~[na:na]
	at java.base/java.util.Collections$UnmodifiableCollection.forEach(Collections.java:1087) ~[na:na]
	at org.comixedproject.batch.comicbooks.writers.AbstractComicWriter.write(AbstractComicWriter.java:48) ~[comixed-batch-1.1.0-1.jar!/:1.1.0-1]
	at org.springframework.batch.core.step.item.SimpleChunkProcessor.writeItems(SimpleChunkProcessor.java:193) ~[spring-batch-core-4.3.4.jar!/:4.3.4]
	at org.springframework.batch.core.step.item.SimpleChunkProcessor.doWrite(SimpleChunkProcessor.java:159) ~[spring-batch-core-4.3.4.jar!/:4.3.4]
	at org.springframework.batch.core.step.item.SimpleChunkProcessor.write(SimpleChunkProcessor.java:294) ~[spring-batch-core-4.3.4.jar!/:4.3.4]
	at org.springframework.batch.core.step.item.SimpleChunkProcessor.process(SimpleChunkProcessor.java:217) ~[spring-batch-core-4.3.4.jar!/:4.3.4]
	at org.springframework.batch.core.step.item.ChunkOrientedTasklet.execute(ChunkOrientedTasklet.java:77) ~[spring-batch-core-4.3.4.jar!/:4.3.4]
	at org.springframework.batch.core.step.tasklet.TaskletStep$ChunkTransactionCallback.doInTransaction(TaskletStep.java:407) ~[spring-batch-core-4.3.4.jar!/:4.3.4]
	at org.springframework.batch.core.step.tasklet.TaskletStep$ChunkTransactionCallback.doInTransaction(TaskletStep.java:331) ~[spring-batch-core-4.3.4.jar!/:4.3.4]
	at org.springframework.transaction.support.TransactionTemplate.execute(TransactionTemplate.java:140) ~[spring-tx-5.3.15.jar!/:5.3.15]
	at org.springframework.batch.core.step.tasklet.TaskletStep$2.doInChunkContext(TaskletStep.java:273) ~[spring-batch-core-4.3.4.jar!/:4.3.4]
	at org.springframework.batch.core.scope.context.StepContextRepeatCallback.doInIteration(StepContextRepeatCallback.java:82) ~[spring-batch-core-4.3.4.jar!/:4.3.4]
	at org.springframework.batch.repeat.support.RepeatTemplate.getNextResult(RepeatTemplate.java:375) ~[spring-batch-infrastructure-4.3.4.jar!/:4.3.4]
	at org.springframework.batch.repeat.support.RepeatTemplate.executeInternal(RepeatTemplate.java:215) ~[spring-batch-infrastructure-4.3.4.jar!/:4.3.4]
	at org.springframework.batch.repeat.support.RepeatTemplate.iterate(RepeatTemplate.java:145) ~[spring-batch-infrastructure-4.3.4.jar!/:4.3.4]
	at org.springframework.batch.core.step.tasklet.TaskletStep.doExecute(TaskletStep.java:258) ~[spring-batch-core-4.3.4.jar!/:4.3.4]
	at org.springframework.batch.core.step.AbstractStep.execute(AbstractStep.java:208) ~[spring-batch-core-4.3.4.jar!/:4.3.4]
	at org.springframework.batch.core.job.SimpleStepHandler.handleStep(SimpleStepHandler.java:152) ~[spring-batch-core-4.3.4.jar!/:4.3.4]
	at org.springframework.batch.core.job.AbstractJob.handleStep(AbstractJob.java:413) ~[spring-batch-core-4.3.4.jar!/:4.3.4]
	at org.springframework.batch.core.job.SimpleJob.doExecute(SimpleJob.java:136) ~[spring-batch-core-4.3.4.jar!/:4.3.4]
	at org.springframework.batch.core.job.AbstractJob.execute(AbstractJob.java:320) ~[spring-batch-core-4.3.4.jar!/:4.3.4]
	at org.springframework.batch.core.launch.support.SimpleJobLauncher$1.run(SimpleJobLauncher.java:149) ~[spring-batch-core-4.3.4.jar!/:4.3.4]
	at java.base/java.lang.Thread.run(Thread.java:831) ~[na:na]
Caused by: org.hibernate.exception.ConstraintViolationException: could not execute statement
	at org.hibernate.exception.internal.SQLExceptionTypeDelegate.convert(SQLExceptionTypeDelegate.java:59) ~[hibernate-core-5.6.4.Final.jar!/:5.6.4.Final]
	at org.hibernate.exception.internal.StandardSQLExceptionConverter.convert(StandardSQLExceptionConverter.java:37) ~[hibernate-core-5.6.4.Final.jar!/:5.6.4.Final]
	at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:113) ~[hibernate-core-5.6.4.Final.jar!/:5.6.4.Final]
	at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:99) ~[hibernate-core-5.6.4.Final.jar!/:5.6.4.Final]
	at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.executeUpdate(ResultSetReturnImpl.java:200) ~[hibernate-core-5.6.4.Final.jar!/:5.6.4.Final]
	at org.hibernate.dialect.identity.GetGeneratedKeysDelegate.executeAndExtract(GetGeneratedKeysDelegate.java:58) ~[hibernate-core-5.6.4.Final.jar!/:5.6.4.Final]
	at org.hibernate.id.insert.AbstractReturningDelegate.performInsert(AbstractReturningDelegate.java:43) ~[hibernate-core-5.6.4.Final.jar!/:5.6.4.Final]
	at org.hibernate.persister.entity.AbstractEntityPersister.insert(AbstractEntityPersister.java:3279) ~[hibernate-core-5.6.4.Final.jar!/:5.6.4.Final]
	at org.hibernate.persister.entity.AbstractEntityPersister.insert(AbstractEntityPersister.java:3885) ~[hibernate-core-5.6.4.Final.jar!/:5.6.4.Final]
	at org.hibernate.action.internal.EntityIdentityInsertAction.execute(EntityIdentityInsertAction.java:84) ~[hibernate-core-5.6.4.Final.jar!/:5.6.4.Final]
	at org.hibernate.engine.spi.ActionQueue.execute(ActionQueue.java:645) ~[hibernate-core-5.6.4.Final.jar!/:5.6.4.Final]
	at org.hibernate.engine.spi.ActionQueue.addResolvedEntityInsertAction(ActionQueue.java:282) ~[hibernate-core-5.6.4.Final.jar!/:5.6.4.Final]
	at org.hibernate.engine.spi.ActionQueue.addInsertAction(ActionQueue.java:263) ~[hibernate-core-5.6.4.Final.jar!/:5.6.4.Final]
	at org.hibernate.engine.spi.ActionQueue.addAction(ActionQueue.java:317) ~[hibernate-core-5.6.4.Final.jar!/:5.6.4.Final]
	at org.hibernate.event.internal.AbstractSaveEventListener.addInsertAction(AbstractSaveEventListener.java:330) ~[hibernate-core-5.6.4.Final.jar!/:5.6.4.Final]
	at org.hibernate.event.internal.AbstractSaveEventListener.performSaveOrReplicate(AbstractSaveEventListener.java:287) ~[hibernate-core-5.6.4.Final.jar!/:5.6.4.Final]
	at org.hibernate.event.internal.AbstractSaveEventListener.performSave(AbstractSaveEventListener.java:193) ~[hibernate-core-5.6.4.Final.jar!/:5.6.4.Final]
	at org.hibernate.event.internal.AbstractSaveEventListener.saveWithGeneratedId(AbstractSaveEventListener.java:123) ~[hibernate-core-5.6.4.Final.jar!/:5.6.4.Final]
	at org.hibernate.event.internal.DefaultMergeEventListener.saveTransientEntity(DefaultMergeEventListener.java:271) ~[hibernate-core-5.6.4.Final.jar!/:5.6.4.Final]
	at org.hibernate.event.internal.DefaultMergeEventListener.entityIsTransient(DefaultMergeEventListener.java:243) ~[hibernate-core-5.6.4.Final.jar!/:5.6.4.Final]
	at org.hibernate.event.internal.DefaultMergeEventListener.onMerge(DefaultMergeEventListener.java:175) ~[hibernate-core-5.6.4.Final.jar!/:5.6.4.Final]
	at org.hibernate.event.service.internal.EventListenerGroupImpl.fireEventOnEachListener(EventListenerGroupImpl.java:118) ~[hibernate-core-5.6.4.Final.jar!/:5.6.4.Final]
	at org.hibernate.internal.SessionImpl.fireMerge(SessionImpl.java:853) ~[hibernate-core-5.6.4.Final.jar!/:5.6.4.Final]
	at org.hibernate.internal.SessionImpl.merge(SessionImpl.java:826) ~[hibernate-core-5.6.4.Final.jar!/:5.6.4.Final]
	at org.hibernate.engine.spi.CascadingActions$6.cascade(CascadingActions.java:261) ~[hibernate-core-5.6.4.Final.jar!/:5.6.4.Final]
	at org.hibernate.engine.internal.Cascade.cascadeToOne(Cascade.java:510) ~[hibernate-core-5.6.4.Final.jar!/:5.6.4.Final]
	at org.hibernate.engine.internal.Cascade.cascadeAssociation(Cascade.java:434) ~[hibernate-core-5.6.4.Final.jar!/:5.6.4.Final]
	at org.hibernate.engine.internal.Cascade.cascadeProperty(Cascade.java:220) ~[hibernate-core-5.6.4.Final.jar!/:5.6.4.Final]
	at org.hibernate.engine.internal.Cascade.cascade(Cascade.java:153) ~[hibernate-core-5.6.4.Final.jar!/:5.6.4.Final]
	at org.hibernate.event.internal.DefaultMergeEventListener.cascadeOnMerge(DefaultMergeEventListener.java:518) ~[hibernate-core-5.6.4.Final.jar!/:5.6.4.Final]
	at org.hibernate.event.internal.DefaultMergeEventListener.entityIsDetached(DefaultMergeEventListener.java:346) ~[hibernate-core-5.6.4.Final.jar!/:5.6.4.Final]
	at org.hibernate.event.internal.DefaultMergeEventListener.onMerge(DefaultMergeEventListener.java:172) ~[hibernate-core-5.6.4.Final.jar!/:5.6.4.Final]
	at org.hibernate.event.internal.DefaultMergeEventListener.onMerge(DefaultMergeEventListener.java:70) ~[hibernate-core-5.6.4.Final.jar!/:5.6.4.Final]
	at org.hibernate.event.service.internal.EventListenerGroupImpl.fireEventOnEachListener(EventListenerGroupImpl.java:107) ~[hibernate-core-5.6.4.Final.jar!/:5.6.4.Final]
	at org.hibernate.internal.SessionImpl.fireMerge(SessionImpl.java:833) ~[hibernate-core-5.6.4.Final.jar!/:5.6.4.Final]
	at org.hibernate.internal.SessionImpl.merge(SessionImpl.java:820) ~[hibernate-core-5.6.4.Final.jar!/:5.6.4.Final]
	at jdk.internal.reflect.GeneratedMethodAccessor307.invoke(Unknown Source) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Method.java:567) ~[na:na]
	at org.springframework.orm.jpa.SharedEntityManagerCreator$SharedEntityManagerInvocationHandler.invoke(SharedEntityManagerCreator.java:311) ~[spring-orm-5.3.15.jar!/:5.3.15]
	at jdk.proxy2/jdk.proxy2.$Proxy161.merge(Unknown Source) ~[na:na]
	at org.springframework.data.jpa.repository.support.SimpleJpaRepository.save(SimpleJpaRepository.java:633) ~[spring-data-jpa-2.6.1.jar!/:2.6.1]
	at jdk.internal.reflect.GeneratedMethodAccessor305.invoke(Unknown Source) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Method.java:567) ~[na:na]
	at org.springframework.data.repository.core.support.RepositoryMethodInvoker$RepositoryFragmentMethodInvoker.lambda$new$0(RepositoryMethodInvoker.java:289) ~[spring-data-commons-2.6.1.jar!/:2.6.1]
	at org.springframework.data.repository.core.support.RepositoryMethodInvoker.doInvoke(RepositoryMethodInvoker.java:137) ~[spring-data-commons-2.6.1.jar!/:2.6.1]
	at org.springframework.data.repository.core.support.RepositoryMethodInvoker.invoke(RepositoryMethodInvoker.java:121) ~[spring-data-commons-2.6.1.jar!/:2.6.1]
	at org.springframework.data.repository.core.support.RepositoryComposition$RepositoryFragments.invoke(RepositoryComposition.java:529) ~[spring-data-commons-2.6.1.jar!/:2.6.1]
	at org.springframework.data.repository.core.support.RepositoryComposition.invoke(RepositoryComposition.java:285) ~[spring-data-commons-2.6.1.jar!/:2.6.1]
	at org.springframework.data.repository.core.support.RepositoryFactorySupport$ImplementationMethodExecutionInterceptor.invoke(RepositoryFactorySupport.java:639) ~[spring-data-commons-2.6.1.jar!/:2.6.1]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.3.15.jar!/:5.3.15]
	at org.springframework.data.repository.core.support.QueryExecutorMethodInterceptor.doInvoke(QueryExecutorMethodInterceptor.java:163) ~[spring-data-commons-2.6.1.jar!/:2.6.1]
	at org.springframework.data.repository.core.support.QueryExecutorMethodInterceptor.invoke(QueryExecutorMethodInterceptor.java:138) ~[spring-data-commons-2.6.1.jar!/:2.6.1]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.3.15.jar!/:5.3.15]
	at org.springframework.data.projection.DefaultMethodInvokingMethodInterceptor.invoke(DefaultMethodInvokingMethodInterceptor.java:80) ~[spring-data-commons-2.6.1.jar!/:2.6.1]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.3.15.jar!/:5.3.15]
	at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:123) ~[spring-tx-5.3.15.jar!/:5.3.15]
	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:388) ~[spring-tx-5.3.15.jar!/:5.3.15]
	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:119) ~[spring-tx-5.3.15.jar!/:5.3.15]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.3.15.jar!/:5.3.15]
	at org.springframework.dao.support.PersistenceExceptionTranslationInterceptor.invoke(PersistenceExceptionTranslationInterceptor.java:137) ~[spring-tx-5.3.15.jar!/:5.3.15]
	... 57 common frames omitted
Caused by: org.h2.jdbc.JdbcSQLIntegrityConstraintViolationException: Unique index or primary key violation: "PUBLIC.CONSTRAINT_A5_INDEX_1 ON PUBLIC.COMICFILEDETAILS(COMICBOOKID) VALUES 319"; SQL statement:
insert into ComicFileDetails (id, ComicBookId, FileHash) values (null, ?, ?) [23505-200]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:459) ~[h2-1.4.200.jar!/:na]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:429) ~[h2-1.4.200.jar!/:na]
	at org.h2.message.DbException.get(DbException.java:205) ~[h2-1.4.200.jar!/:na]
	at org.h2.message.DbException.get(DbException.java:181) ~[h2-1.4.200.jar!/:na]
	at org.h2.index.BaseIndex.getDuplicateKeyException(BaseIndex.java:103) ~[h2-1.4.200.jar!/:na]
	at org.h2.mvstore.db.MVSecondaryIndex.checkUnique(MVSecondaryIndex.java:221) ~[h2-1.4.200.jar!/:na]
	at org.h2.mvstore.db.MVSecondaryIndex.add(MVSecondaryIndex.java:196) ~[h2-1.4.200.jar!/:na]
	at org.h2.mvstore.db.MVTable.addRow(MVTable.java:531) ~[h2-1.4.200.jar!/:na]
	at org.h2.command.dml.Insert.insertRows(Insert.java:195) ~[h2-1.4.200.jar!/:na]
	at org.h2.command.dml.Insert.update(Insert.java:151) ~[h2-1.4.200.jar!/:na]
	at org.h2.command.CommandContainer.executeUpdateWithGeneratedKeys(CommandContainer.java:272) ~[h2-1.4.200.jar!/:na]
	at org.h2.command.CommandContainer.update(CommandContainer.java:191) ~[h2-1.4.200.jar!/:na]
	at org.h2.command.Command.executeUpdate(Command.java:251) ~[h2-1.4.200.jar!/:na]
	at org.h2.jdbc.JdbcPreparedStatement.executeUpdateInternal(JdbcPreparedStatement.java:191) ~[h2-1.4.200.jar!/:na]
	at org.h2.jdbc.JdbcPreparedStatement.executeUpdate(JdbcPreparedStatement.java:152) ~[h2-1.4.200.jar!/:na]
	at com.zaxxer.hikari.pool.ProxyPreparedStatement.executeUpdate(ProxyPreparedStatement.java:61) ~[HikariCP-4.0.3.jar!/:na]
	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeUpdate(HikariProxyPreparedStatement.java) ~[HikariCP-4.0.3.jar!/:na]
	at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.executeUpdate(ResultSetReturnImpl.java:197) ~[hibernate-core-5.6.4.Final.jar!/:5.6.4.Final]
```

--- a/comixed-adaptors/src/main/java/org/comixedproject/adaptors/comicbooks/ComicBookAdaptor.java
+++ b/comixed-adaptors/src/main/java/org/comixedproject/adaptors/comicbooks/ComicBookAdaptor.java
@@ -93,7 +93,7 @@ public class ComicBookAdaptor {
         final ComicArchiveEntry entry = entries.get(index);
         log.trace("Loading entry content: {}", entry.getFilename());
         final byte[] content = archiveAdaptor.readEntry(readHandle, entry.getFilename());
-        log.trace("Getting content adaptor");
+        log.debug("Getting content adaptor for entry: {}", entry.getFilename());
         final ContentAdaptor adaptor = this.fileTypeAdaptor.getContentAdaptorFor(content);
         if (adaptor != null) {
           log.trace("Invoking content adaptor");
@@ -283,6 +283,8 @@ public class ComicBookAdaptor {
           log.trace("File content found");
           break;
         }
+        // reset the result value
+        result = null;
       }
       log.trace("Closing comic book file");
       archiveAdaptor.closeArchiveForRead(readHandle);

--- a/comixed-batch/src/main/java/org/comixedproject/batch/comicbooks/processors/LoadFileDetailsProcessor.java
+++ b/comixed-batch/src/main/java/org/comixedproject/batch/comicbooks/processors/LoadFileDetailsProcessor.java
@@ -40,9 +40,12 @@ public class LoadFileDetailsProcessor implements ItemProcessor<ComicBook, ComicB
 
   @Override
   public ComicBook process(final ComicBook comicBook) {
-    log.debug("Creating file details container: id={}", comicBook.getId());
-    final ComicFileDetails fileDetails = new ComicFileDetails(comicBook);
-    comicBook.setFileDetails(fileDetails);
+    ComicFileDetails fileDetails = comicBook.getFileDetails();
+    if (fileDetails == null) {
+      log.debug("Creating file details container: id={}", comicBook.getId());
+      fileDetails = new ComicFileDetails(comicBook);
+      comicBook.setFileDetails(fileDetails);
+    }
     log.trace("Getting comicBook file hash");
     try {
       fileDetails.setHash(

--- a/comixed-batch/src/test/java/org/comixedproject/batch/comicbooks/processors/LoadFileDetailsProcessorTest.java
+++ b/comixed-batch/src/test/java/org/comixedproject/batch/comicbooks/processors/LoadFileDetailsProcessorTest.java
@@ -38,6 +38,7 @@ public class LoadFileDetailsProcessorTest {
   @InjectMocks private LoadFileDetailsProcessor processor;
   @Mock private GenericUtilitiesAdaptor genericUtilitiesAdaptor;
   @Mock private ComicBook comicBook;
+  @Mock private ComicFileDetails fileDetails;
 
   @Captor private ArgumentCaptor<ComicFileDetails> comicFileDetailsArgumentCaptor;
 
@@ -74,5 +75,20 @@ public class LoadFileDetailsProcessorTest {
     assertNotNull(fileDetails);
     assertSame(comicBook, fileDetails.getComicBook());
     assertEquals(TEST_HASH, fileDetails.getHash());
+  }
+
+  @Test
+  public void testProcessHasFileDetails() throws Exception {
+    Mockito.when(comicBook.getFilename()).thenReturn(TEST_COMIC_FILENAME);
+    Mockito.when(comicBook.getFileDetails()).thenReturn(fileDetails);
+    Mockito.when(genericUtilitiesAdaptor.createHash(Mockito.any(InputStream.class)))
+        .thenReturn(TEST_HASH);
+
+    final ComicBook result = processor.process(comicBook);
+
+    assertNotNull(result);
+    assertSame(comicBook, result);
+
+    Mockito.verify(comicBook, Mockito.never()).setFileDetails(Mockito.any());
   }
 }

